extensions/vscode/src/eventErrors.test.ts

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2024 by Posit Software, PBC.
+
+import { describe, expect, test } from "vitest";
+import { EventStreamMessage } from "./api";
+import {
+  EventStreamMessageErrorCoded,
+  isCodedEventErrorMessage,
+  isEvtErrTargetNotFound,
+  isEvtErrTargetForbidden,
+  handleEventCodedError,
+} from "./eventErrors";
+import { ErrorCode } from "./utils/errorTypes";
+
+const mkEventStreamMsg = (
+  data = {},
+  errCode?: ErrorCode,
+): EventStreamMessage => {
+  return {
+    type: "publish/failure",
+    time: "Tue Oct 01 2024 10:00:00 GMT-0600",
+    data,
+    errCode,
+    error: "failed to publish",
+  };
+};
+
+describe("Event errors", () => {
+  test("isCodedEventErrorMessage", () => {
+    // Message without error code
+    let streamMsg = mkEventStreamMsg();
+    let result = isCodedEventErrorMessage(streamMsg);
+    expect(result).toBe(false);
+
+    // Message with error code
+    streamMsg = mkEventStreamMsg({}, "deploymentTargetNotFound");
+    result = isCodedEventErrorMessage(streamMsg);
+    expect(result).toBe(true);
+  });
+
+  test("isEvtErrTargetNotFound", () => {
+    // Message with another error code
+    let streamMsg = mkEventStreamMsg({}, "unknown");
+    let result = isEvtErrTargetNotFound(
+      streamMsg as EventStreamMessageErrorCoded,
```

Comment:
The `as` keyword always makes me nervous. This is test code so feel free to ignore this, but I would make `mkEventStreamMsg` have a generic so you could do something like

```
let streamMsg = mkEventStreamMsg<EventStreamMessageErrorCoded>({}, "unknown");
```

or similar. That would remove the need for the `as`. You could even change the return type if `errCode` is passed by overloading the definition of the method.

Reply:
That's a great idea, I'll try that.